### PR TITLE
[Merged by Bors] - start syncing after genesis layer

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

simplify mesh's layer processing logic

## Changes

- set processed layer to genesis layer when initializing mesh. this means syncing will start from genesis+1 layer
- remove logic for out-of-order validated layer in mesh as this cannot happen with tortoise logic
- use gomock for state/tortoise in mesh tests
- in mesh, change rewards calculations to use utility functions

## Test Plan

unit test

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by countvonzero** [comment]
bors try

File: mesh/mesh.go [hunk 1]
```diff
@@ -641,7 +594,7 @@ func (msh *Mesh) getAggregatedLayerHash(layerID types.LayerID) (types.Hash32, er
 	return hash, fmt.Errorf("get from DB: %w", err)
 }
 
-func (msh *Mesh) extractUniqueOrderedTransactions(l *types.Layer) (validBlockTxs []*types.Transaction) {
+func extractUniqueOrderedTransactions(l *types.Layer, mdb *DB, logger log.Log) (validBlockTxs []*types.Transaction) {
```
Comment (dshulyak) [inline mesh/mesh.go]: why do you prefer to pass logger as a last value? i think we need some consistent approach

i am usually using it as a first argument as to me the idea is the same as with ctx. so logger logically can be a part of context

**Review by dshulyak — APPROVED** [review]

File: mesh/mesh.go (hunk 1 above)
Comment (countvonzero) [reply mesh/mesh.go]: i don't actually have a preference. changed.

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Review by nkryuchkov — APPROVED** [review]

**Pull request closed without merging**